mm2src/mm2_main/src/lp_swap/swap_v2_rpcs.rs

```diff
@@ -501,3 +503,170 @@ pub(crate) async fn active_swaps_rpc(
         statuses,
     })
 }
+
+#[derive(Deserialize)]
+pub(crate) struct StopSwapRequest {
+    uuid: Uuid,
+}
+
+#[derive(Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub(crate) enum StopSwapErr {
+    Internal(String),
+    NotRunning,
+}
+
+impl HttpStatusCode for StopSwapErr {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            StopSwapErr::Internal(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            StopSwapErr::NotRunning => StatusCode::BAD_REQUEST,
+        }
+    }
+}
+
+#[derive(Serialize)]
+pub(crate) struct StopSwapResponse {
+    result: String,
+}
+
+pub(crate) async fn stop_swap_rpc(ctx: MmArc, req: StopSwapRequest) -> MmResult<StopSwapResponse, StopSwapErr> {
```

Comment:
I am a bit confused: this stop_swap_rpc is for legacy swaps, why is it in the swap_v2_rpc.rs file?

Reply:
i actually wanted to rant about this but forgot xD

i came to a conclusion that the `v2` here is a reference to the `rpc` (`dispatcher.rs` vs `dispatcher_legacy.rs`) and not to the `swap`, this is because other rpcs in this file still handle legacy stuff and there are other similar ones like these in `lp_swap.rs` that are used in `dispatcher_legacy.rs` and they also handle tpv2 stuff. this only means that the `v2` here means rpc v2 :)

why the file isn't named `swap_rpcs_v2.rs`. no idea!

Reply:
Are we going to add stop swap for TPU swaps as well?

Reply:
I think we eventually will. as we will apply a similar fix (endless recovery trials) to tpv2 as well.

Reply:
> Are we going to add stop swap for TPU swaps as well?

we are going to add refund or spend functionality to TPU if swap faced error, stop rpc allows user to stop such process.

I think in tpu we should allow user to be able to call stop rpc only if swap process go to `_FundingRefundRequired` state

Reply:
i d like to make this resolved.
there is FundingRefundOrSpendRequired suggestion though

Reply:
there is no `FundingRefundOrSpendRequired` in legacy swaps. though i wanna change the `refund` notion in events like `TakerPaymentRefundStarted` and commands like `RefundTakerPayment` to `recover` but that would create a backward compatibility issues with loading older swaps so wanna discuss first how could this be done neatly.

Reply:
Ah okay it was about TPU states